specs/accordion.md

```diff
@@ -0,0 +1,184 @@
+
+- On April 16th Randy wakes up in a cold sweat because he realized he made a mistake in filing his taxes with the IRS. He opens his laptop and goes to the IRS website and sees a link to "frequently asked questions". Randy sees a list of questions on the page with an arrow next to it indicating that answers are hidden. He sees a question titled, "What should I do if I made a mistake on my federal turn that I've already filed?" Randy clicks the question and sees a section detailing out the exact steps to take. With his anxiety eased, Randy goes back to bed and falls asleep.
+
+### Features
+
+1. Optionally allow only a single section to be expanded. This behavior is opt-in and not default behavior.
+
+2. Support for interactive content within the header, such as a menu button, checkbox, etc.
+
+3. Optional support for nested accordions. Ideally this behavior will "just work" and no special behaviors will need to be added to enable/support this. The expectation here should be that an accordion takes content and whatever content is inside the accordion panel respects its own interaction model.
+
+### Prior Art/Examples
+
+- [Ant Design](https://ant.design/components/collapse/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/accordion/usage/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/accordion/)
+- [Fluent UI](https://fluentsite.z22.web.core.windows.net/components/accordion/accessibility)
+- [Primer](https://primer.style/components/Details)
+- [Bootstrap](https://getbootstrap.com/docs/4.3/components/collapse/)
+
+---
+
+### API
+
+**Accordion**
+*Component Name*
+- `fast-accordion`
+
+*Attributes:*
+- `expandmode` - enum
+  - single
+  - multi - default
+
+*Events*
+- `change: CustomEvent`
+  - no custom data
+  - bubbles
+
+**Accordion Item**
+*Component names:*
+- `fast-accordion-item`
+
+*Attributes:*
+- `expanded` - boolean
+- `id` - string
+- `headinglevel` - enum
+  - 1
+  - 2
+  - 3 - default
+  - 4
+  - 5
+  - 6
+
+*Parts:*
+- panel
+- button
+
+*Slot Names*
+- default
+- heading
+- glyph
+
+### Anatomy and Appearance
+
+```HTML
+<!-- shadow root -->
+<div>
+    <div role="heading" aria-level="3">
+        <button
+            type="button"
+            class="button"
+            part="button"
+            aria-expanded="true"
+            aria-controls="accordion1-panel"
+            id="accordion1"
+        >
+            <slot name="heading" part="heading">Panel one</slot>
+            <slot name="glyph" part="glyph"></slot>
```

Comment:
Accordians support (and we have had several designs within MSFT) additional interactive elements (more menu, checkbox) within the header. How can we light up support for those if I want a checkbox before the header content and a more menu after?

Reply:
The heading slot can take any elements you want to pass to it